### PR TITLE
[CLI-2182] Add `--key-schema` flag to `kafka topic produce`

### PR DESCRIPTION
**Pull request by brianstrauch**
Release Notes
-------------
New Features
- Add `--key-format`, `--key-schema`, and `--key-references` to `kafka topic produce`
- Add `--key-format` to `kafka topic consume`

Bug Fixes
- PLACEHOLDER

Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----
Add the ability to produce and consume messages with a serialized key

Test & Review
-------------
Manually tested, since there are no tests for produce/consume

### CODE REVIEW

File: internal/pkg/cmd/flags.go [hunk 1]
```diff
@@ -396,6 +396,17 @@ func AddTypeFlag(cmd *cobra.Command) {
 	RegisterFlagCompletionFunc(cmd, "type", func(_ *cobra.Command, _ []string) []string { return kafka.Types })
 }
 
+func AddKeyFormatFlag(cmd *cobra.Command) {
+	arr := []string{"string", "avro", "jsonschema", "protobuf"}
+	str := utils.ArrayToCommaDelimitedString(arr, "or")
+
+	cmd.Flags().String("key-format", "string", fmt.Sprintf("Format of message key as %s. Note that schema references are not supported for Avro.", str))
```
Comment (sgagniere) [inline internal/pkg/cmd/flags.go:403]: It's not immediately clear to me what `Format of message key as %s` means.

File: internal/cmd/kafka/confluent_kafka.go [hunk 2]
```diff
@@ -183,54 +184,72 @@ func GetRebalanceCallback(offset ckafka.Offset, partitionFilter PartitionFilter)
 	}
 }
 
-func consumeMessage(e *ckafka.Message, h *GroupHandler) error {
-	value := e.Value
+func consumeMessage(message *ckafka.Message, h *GroupHandler) error {
+	// TODO: Deserialize key
```
Comment (sgagniere) [inline internal/cmd/kafka/confluent_kafka.go:188]: Is this a stale comment?

Comment (brianstrauch) [reply internal/cmd/kafka/confluent_kafka.go:188]:
```suggestion
```

**Review by sgagniere — APPROVED** [review]

File: internal/cmd/schema-registry/schema_utilities.go [hunk 3]
```diff
@@ -24,15 +24,15 @@ type registerSchemaResponse struct {
...
-	ValueFormat string
-	SchemaType  string
-	SchemaPath  string
-	Refs        []srsdk.SchemaReference
-	Metadata    *srsdk.Metadata
-	Ruleset     *srsdk.RuleSet
-	Normalize   bool
+	SchemaDir  string
+	Subject    string
+	Format     string
```
Comment (MuweiHe) [inline internal/cmd/schema-registry/schema_utilities.go:29]: a little confused here: does this format refer to key or value?

File: internal/cmd/kafka/command_topic_produce_onprem.go [hunk 4]
```diff
@@ -115,20 +129,35 @@ func (c *command) produceOnPrem(cmd *cobra.Command, args []string) error {
...
+		SchemaType: valueSerializer.GetSchemaName(),
+		Format:     valueFormat,
+		SchemaPath: schema,
+		Refs:       refs,
+	}
+	valueMetaInfo, referencePathMap, err := c.registerSchemaOnPrem(cmd, valueSchemaConfigs)
+	if err != nil {
+		return err
+	}
+	if err := valueSerializer.LoadSchema(schema, referencePathMap); err != nil {
```
Comment (MuweiHe) [inline internal/cmd/kafka/command_topic_produce_onprem.go:160]: may refer this as "valueSchema"

File: internal/cmd/kafka/command_topic_produce.go [hunk 5]
```diff
@@ -31,7 +32,9 @@ func (c *command) newProduceCommand() *cobra.Command {
 		Annotations:       map[string]string{pcmd.RunRequirement: pcmd.RequireCloudLogin},
 	}
 
+	cmd.Flags().String("key-schema", "", "The ID or filepath of the message key schema.")
 	cmd.Flags().String("schema", "", "The ID or filepath of the message value schema.")
+	pcmd.AddKeyFormatFlag(cmd)
 	pcmd.AddValueFormatFlag(cmd)
 	cmd.Flags().String("references", "", "The path to the references file.")
```
Comment (MuweiHe) [inline internal/cmd/kafka/command_topic_produce.go:39]: the reference file is for value schema i suppose? may need to update the description

File: test/fixtures/output/kafka/topic/produce-help.golden [hunk 6]
```diff
@@ -6,7 +6,9 @@ Usage:
   confluent kafka topic produce <topic> [flags]
 
 Flags:
+      --key-schema string                   The ID or filepath of the message key schema.
       --schema string                       The ID or filepath of the message value schema.
+      --key-format string                   Format of message key as "string", "avro", "jsonschema", or "protobuf". Note that schema references are not supported for Avro. (default "string")
```
Comment (MuweiHe) [inline test/fixtures/output/kafka/topic/produce-help.golden:11]: does key schema have reference files?

File: internal/cmd/schema-registry/schema_utilities.go (hunk 3 above)
Comment (brianstrauch) [reply internal/cmd/schema-registry/schema_utilities.go:29]: Either! It's used for both.

File: internal/cmd/kafka/command_topic_produce_onprem.go (hunk 4 above)
Comment (brianstrauch) [reply internal/cmd/kafka/command_topic_produce_onprem.go:160]: I was calling it "schema" to match `--schema`. In v4 I want to change this flag to `--value-schema` though!

File: internal/cmd/kafka/command_topic_produce.go (hunk 5 above)
Comment (brianstrauch) [reply internal/cmd/kafka/command_topic_produce.go:39]: Ah, didn't think of this. We might need a `--key-references`?

Comment (MuweiHe) [reply internal/cmd/kafka/command_topic_produce.go:39]: may confirm with dg team. not sure is it's supported

Comment (brianstrauch) [reply internal/cmd/kafka/command_topic_produce.go:39]: Key serialization should be the same exact process as value serialization, so if it's supported for one it should be supported for both!

File: test/fixtures/output/kafka/topic/produce-help.golden (hunk 6 above)
Comment (brianstrauch) [reply test/fixtures/output/kafka/topic/produce-help.golden:11]: Added `--key-references`!

**Pull request merged**